### PR TITLE
Remove in-tree bounded types and use bounded-collections crate

### PR DESCRIPTION
**Pull request by KiChjang**
This PR removes the in-tree bounded types and instead imports the `bounded-collections` crate and re-exports everything under it in `sp-core`.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2318831

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2318884

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2323735

File: primitives/core/src/lib.rs [hunk 1]
```diff
@@ -81,6 +80,8 @@ pub use self::hasher::blake2::Blake2Hasher;
 pub use self::hasher::keccak::KeccakHasher;
 pub use hash_db::Hasher;
 
+pub use bounded_collections as bounded;
+pub use bounded_collections::*;
```
Comment (bkchr) [inline primitives/core/src/lib.rs:84]: Before `BoundedVec` etc also wasn't exported in the top level of the crate. We should probably change this again?

**Review by bkchr — APPROVED** [review]

File: primitives/core/src/lib.rs (hunk 1 above)
Comment (KiChjang) [reply primitives/core/src/lib.rs:84]: A couple of types were in fact exported from the top level, and those were the types that aren't strictly related to the bounded types, e.g. `TypedGet`, `Get`, `TryCollect`, etc. Should I export only these types from the top level then?

**Review by ggwpez — APPROVED** [review]

**Review by mrcnski — APPROVED** [review]

File: primitives/core/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply primitives/core/src/lib.rs:84]: I would say yes.

**Comment by KiChjang** [comment]
bot merge

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2330798

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2330797

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2330786

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2330787

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2330788

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2330793

**Comment by KiChjang** [comment]
bot merge

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**